openerp/addons/base/res/res_partner.py

```diff
@@ -538,6 +548,18 @@ def write(self, vals):
             self._fields_sync(partner, vals)
         return result
 
+    #Gravatar Image
+    def _get_gravatar_image(self, email):
+        email_hash = hashlib.md5(email.lower()).hexdigest()
+        url = "https://www.gravatar.com/avatar/" + email_hash
+        try:
+            image_content = urllib2.urlopen(url + "?d=404&s=128", timeout=5).read()
```

Comment:
It will check for 5 seconds only
If gravatar not found then it will timeout within 5 seconds